starters/express-typescript-typeorm-postgres/src/main.ts

```diff
@@ -2,11 +2,26 @@
 require('dotenv').config();
 import { bootstrapApp } from './bootstrap-app';
 import { dataSource } from './datasource';
+import cors from 'cors';
 
 const PORT = process.env.PORT || 3333;
 
 const app = bootstrapApp();
 
+const whitelist: string[] = JSON.parse(process.env.CORS_ALLOWED_ORIGINS || []);
+
+const corsOptions = {
+  origin: (origin, callback) => {
+    if (whitelist.indexOf(origin) !== -1) {
```

Comment:
Please elaborate:

If the allowedOrigins variable is an empty array, should we throw an error always? I think in that case we should allow same-origin requests to go through. Does this implementation enable that? 🤔